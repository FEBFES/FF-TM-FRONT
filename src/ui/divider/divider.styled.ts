import styled from 'styled-components';
import { directionType } from './divider';

export const SDivider = styled.div<{ direction: directionType }>`
  background: var(--bg-border);
  width: ${({ direction }) => (direction === 'col' ? '2px' : '80%')}
  height: ${({ direction }) => (direction === 'col' ? '80%' : '2px')}
`;
