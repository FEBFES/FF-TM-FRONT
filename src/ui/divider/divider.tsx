import React from 'react';
import { SDivider } from './divider.styled';

export type directionType = 'row' | 'col';

interface DividerProps {
  direction: directionType;
}

export const Divider: React.FC<DividerProps> = ({
  direction = 'row',
}): JSX.Element => {
  return <SDivider direction={direction} />;
};
