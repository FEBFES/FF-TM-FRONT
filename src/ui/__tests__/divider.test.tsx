import React from 'react';
import {Divider} from '../divider/divider';
import renderer from 'react-test-renderer';

describe('<Divider />', () => {
    it('SNAPSHOTS: UI - Divider (col) component - render', () => {
        const elem = renderer.create(<Divider direction='col' />).toJSON();
        expect(elem).toMatchSnapshot();
    });

    it('SNAPSHOTS: UI - Divider (row) component - render', () => {
        const elem = renderer.create(<Divider direction='row' />).toJSON();
        expect(elem).toMatchSnapshot();
    });
})