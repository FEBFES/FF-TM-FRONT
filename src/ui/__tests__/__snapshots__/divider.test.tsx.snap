// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Divider /> SNAPSHOTS: UI - Divider (col) component - render 1`] = `
<div
  className="sc-bcPKhP kLuxxo"
  direction="col"
/>
`;

exports[`<Divider /> SNAPSHOTS: UI - Divider (row) component - render 1`] = `
<div
  className="sc-bcPKhP jwnRNo"
  direction="row"
/>
`;
